gojek/feast
# Feast 0.3 Continuous Integration (CI) Update

Current CI script using Prow is working correctly only for Feast 0.1.
In Feast 0.3, there are updates to the dependencies and steps required to build and test different Feast components.

This is a pull request that update the test and build scripts for Feast 0.3. 
Resolves #275 

The CI stages performed are the following  
(check `.prow/scripts/` folder and `.prow/config.yaml` for details):
1. Test Feast Core and Ingestion
2. Test Feast Serving
3. Test Java SDK
4. Test Python SDK
5. Test Golang SDK
6. Test end-to-end flow
   1. Setup Source (Kafka), Feast metadata DB (Postgres), Feast Store (Redis) in a test instance
   2. Build and start Feast Core and Serving
   3. Use Python SDK to register FeatureSet and perform ingestion
   4. Validate the FeatureSet is successfully registered and features are ingested correctly

NOTE:  
This is likely to cause current tests in master to fail. This is expected.
